error: no module named 'c_builtins' available within module

I get this when translating C code that uses compiler builtins:
```c
void foo(void) {
    int x;
    __builtin___memset_chk (&x, 0, sizeof(x), __builtin_object_size (&x, 0));
}

int main(void) {
    foo();
}
```

produces:
```zig
pub const __builtin = @import("c_builtins");
pub const __helpers = @import("helpers");

pub export fn foo() void {
    var x: c_int = undefined;
    _ = &x;
    _ = __builtin.memset_chk(@ptrCast(@alignCast(&x)), 0, @sizeOf(@TypeOf(x)), __builtin.object_size(@ptrCast(@alignCast(&x)), 0));
}
pub export fn main() c_int {
    foo();
    return 0;
}
```

```
out.zig:1:31: error: no module named 'c_builtins' available within module 'out'
pub const __builtin = @import("c_builtins");
                              ^~~~~~~~~~~~
```

This is using translate-c as a standalone executable. Zig version `0.15.0-dev.847+850655f06`

```
ehaas@Mac translate-c % cat test.c
void foo(void) {
    int x;
    __builtin___memset_chk (&x, 0, sizeof(x), __builtin_object_size (&x, 0));
}

int main(void) {
    foo();
}
```

```
ehaas@Mac translate-c % ./zig-out/bin/translate-c test.c > out.zig
ehaas@Mac translate-c % zig run out.zig
out.zig:1:31: error: no module named 'c_builtins' available within module 'out'
pub const __builtin = @import("c_builtins");
                              ^~~~~~~~~~~~
referenced by:
    foo: out.zig:7:9
    foo: out.zig:4:5
    4 reference(s) hidden; use '-freference-trace=6' to see all references
```

--- a/build.zig
+++ b/build.zig
@@ -42,6 +42,11 @@ pub fn build(b: *std.Build) void {
         .install_dir = .prefix,
         .install_subdir = "include",
     });
+    b.installDirectory(.{
+        .source_dir = b.path("lib"),
+        .install_dir = .prefix,
+        .install_subdir = "lib",
+    });
     b.installArtifact(translate_c_exe);
 
     // Re-expose the path to Aro's "resource directory" (which is actually just the repo root). This
@@ -66,7 +71,7 @@ pub fn build(b: *std.Build) void {
         break :step &run_cmd.step;
     });
 
-    const fmt_dirs: []const []const u8 = &.{ "build", "src", "test" };
+    const fmt_dirs: []const []const u8 = &.{ "build", "build.zig", "src", "lib", "test", "examples" };
 
     const fmt_step = b.step("fmt", "Modify source files in place to have conforming formatting");
     fmt_step.dependOn(&b.addFmt(.{ .paths = fmt_dirs }).step);

--- a/build/Translator.zig
+++ b/build/Translator.zig
@@ -46,6 +46,7 @@ pub const Options = struct {
     c_source_file: Build.LazyPath,
     target: Build.ResolvedTarget,
     optimize: std.builtin.OptimizeMode,
+    module_libs: bool = true,
     link_libc: bool = true,
     warnings: enum { ignore, show, @"error" } = .ignore,
 };
@@ -117,6 +118,10 @@ pub fn initInner(
     }
 
     appendIncludeArg(run, "-resource-dir", tc_conf.aro_resource_dir);
+
+    if (options.module_libs) {
+        run.addArg("-fmodule-libs");
+    }
 
     return .{
         .output_file = output_file,

--- a/src/Translator.zig
+++ b/src/Translator.zig
@@ -164,12 +164,16 @@ fn warn(t: *Translator, scope: *Scope, tok_idx: TokenIndex, comptime format: []c
     try scope.appendNode(try ZigTag.warning.create(t.arena, value));
 }
 
-pub fn translate(
+pub const Options = struct {
     gpa: mem.Allocator,
     comp: *aro.Compilation,
     pp: *const aro.Preprocessor,
     tree: *const aro.Tree,
-) ![]u8 {
+    module_libs: bool,
+};
+
+pub fn translate(options: Options) ![]u8 {
+    const gpa = options.gpa;
     var arena_allocator = std.heap.ArenaAllocator.init(gpa);
     defer arena_allocator.deinit();
     const arena = arena_allocator.allocator();
@@ -179,9 +183,9 @@ pub fn translate(
         .arena = arena,
         .alias_list = .empty,
         .global_scope = try arena.create(Scope.Root),
-        .comp = comp,
-        .pp = pp,
-        .tree = tree,
+        .comp = options.comp,
+        .pp = options.pp,
+        .tree = options.tree,
     };
     translator.global_scope.* = Scope.Root.init(&translator);
     defer {
@@ -216,12 +220,21 @@ pub fn translate(
     var buf: std.ArrayList(u8) = .init(gpa);
     defer buf.deinit();
 
-    try buf.appendSlice(
-        \\pub const __builtin = @import("c_builtins");
-        \\pub const __helpers = @import("helpers");
-        \\
-        \\
-    );
+    if (options.module_libs) {
+        try buf.appendSlice(
+            \\pub const __builtin = @import("c_builtins");
+            \\pub const __helpers = @import("helpers");
+            \\
+            \\
+        );
+    } else {
+        try buf.appendSlice(
+            \\pub const __builtin = @import("c_builtins.zig");
+            \\pub const __helpers = @import("helpers.zig");
+            \\
+            \\
+        );
+    }
 
     var zig_ast = try ast.render(gpa, translator.global_scope.nodes.items);
     defer {

--- a/src/main.zig
+++ b/src/main.zig
@@ -71,9 +71,11 @@ pub fn main() u8 {
 pub const usage =
     \\Usage {s}: [options] file [CC options]
     \\
-    \\General options:
-    \\  -h, --help      Print this message.
-    \\  -v, --version   Print translate-c version.
+    \\Options:
+    \\  --help              Print this message
+    \\  --version           Print translate-c version
+    \\  -fmodule-libs       Import libraries as modules
+    \\  -fno-module-libs    (default) Install libraries next to output file
     \\
     \\
 ;
@@ -86,23 +88,29 @@ fn translate(d: *aro.Driver, tc: *aro.Toolchain, args: [][:0]u8) !void {
 
     try macro_buf.appendSlice("#define __TRANSLATE_C__ 1\n");
 
+    var module_libs = false;
+
     const aro_args = args: {
         var i: usize = 0;
         for (args) |arg| {
             args[i] = arg;
-            if (mem.eql(u8, arg, "-h") or mem.eql(u8, arg, "--help")) {
+            if (mem.eql(u8, arg, "--help")) {
                 const stdout = std.io.getStdOut();
                 stdout.writer().print(usage, .{args[0]}) catch |er| {
                     return d.fatal("unable to print usage: {s}", .{aro.Driver.errorDescription(er)});
                 };
                 return;
-            } else if (mem.eql(u8, arg, "-v") or mem.eql(u8, arg, "--version")) {
+            } else if (mem.eql(u8, arg, "--version")) {
                 const stdout = std.io.getStdOut();
                 // TODO add version
                 stdout.writeAll("0.0.0-dev\n") catch |er| {
                     return d.fatal("unable to print version: {s}", .{aro.Driver.errorDescription(er)});
                 };
                 return;
+            } else if (mem.eql(u8, arg, "-fmodule-libs")) {
+                module_libs = true;
+            } else if (mem.eql(u8, arg, "-fno-module-libs")) {
+                module_libs = false;
             } else {
                 i += 1;
             }
@@ -147,7 +155,13 @@ fn translate(d: *aro.Driver, tc: *aro.Toolchain, args: [][:0]u8) !void {
         return error.FatalError;
     }
 
-    const rendered_zig = try Translator.translate(gpa, d.comp, &pp, &c_tree);
+    const rendered_zig = try Translator.translate(.{
+        .gpa = gpa,
+        .comp = d.comp,
+        .pp = &pp,
+        .tree = &c_tree,
+        .module_libs = module_libs,
+    });
     defer gpa.free(rendered_zig);
 
     var close_out_file = false;
@@ -157,6 +171,10 @@ fn translate(d: *aro.Driver, tc: *aro.Toolchain, args: [][:0]u8) !void {
 
     if (d.output_name) |path| blk: {
         if (std.mem.eql(u8, path, "-")) break :blk;
+        if (std.fs.path.dirname(path)) |dirname| {
+            std.fs.cwd().makePath(dirname) catch |err|
+                return d.fatal("failed to create path to '{s}': {s}", .{ path, aro.Driver.errorDescription(err) });
+        }
         out_file = std.fs.cwd().createFile(path, .{}) catch |err| {
             return d.fatal("failed to create output file '{s}': {s}", .{ path, aro.Driver.errorDescription(err) });
         };
@@ -166,7 +184,45 @@ fn translate(d: *aro.Driver, tc: *aro.Toolchain, args: [][:0]u8) !void {
     out_file.writeAll(rendered_zig) catch |err|
         return d.fatal("failed to write result to '{s}': {s}", .{ out_file_path, aro.Driver.errorDescription(err) });
 
+    if (!module_libs) {
+        const dest_path = if (d.output_name) |path| std.fs.path.dirname(path) else null;
+        installLibs(d, dest_path) catch |err|
+            return d.fatal("failed to install library files: {s}", .{aro.Driver.errorDescription(err)});
+    }
+
     if (fast_exit) process.exit(0);
+}
+
+fn installLibs(d: *aro.Driver, dest_path: ?[]const u8) !void {
+    const gpa = d.comp.gpa;
+    const cwd = std.fs.cwd();
+
+    const self_exe_path = try std.fs.selfExePathAlloc(gpa);
+    defer gpa.free(self_exe_path);
+
+    var cur_dir: []const u8 = self_exe_path;
+    while (std.fs.path.dirname(cur_dir)) |dirname| : (cur_dir = dirname) {
+        var base_dir = cwd.openDir(dirname, .{}) catch continue;
+        defer base_dir.close();
+
+        var lib_dir = base_dir.openDir("lib", .{}) catch continue;
+        defer lib_dir.close();
+
+        lib_dir.access("c_builtins.zig", .{}) catch continue;
+
+        {
+            const install_path = try std.fs.path.join(gpa, &.{ dest_path orelse "", "c_builtins.zig" });
+            defer gpa.free(install_path);
+            try lib_dir.copyFile("c_builtins.zig", cwd, install_path, .{});
+        }
+        {
+            const install_path = try std.fs.path.join(gpa, &.{ dest_path orelse "", "helpers.zig" });
+            defer gpa.free(install_path);
+            try lib_dir.copyFile("helpers.zig", cwd, install_path, .{});
+        }
+        return;
+    }
+    return error.FileNotFound;
 }
 
 comptime {

